src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesSynchronisationService.kt

```diff
@@ -27,28 +29,41 @@ class CaseNotesSynchronisationService(
   }
 
   suspend fun caseNoteInserted(event: CaseNotesEvent) {
+    val telemetry = event.toTelemetryProperties()
+
     if (event.auditModuleName == "DPS_SYNCHRONISATION") {
-      telemetryClient.trackEvent("casenotes-synchronisation-skipped", event.toTelemetryProperties())
+      telemetryClient.trackEvent("casenotes-synchronisation-skipped", telemetry)
       return
     }
     val nomisCaseNote = nomisApiService.getCaseNote(event.caseNoteId)
     caseNotesService.upsertCaseNote(nomisCaseNote.toDPSCreateSyncCaseNote()).apply {
       tryToCreateCaseNoteMapping(
         event,
         this.caseNoteId!!,
-      )
+      ).also { mappingCreateResult ->
+        val mappingSuccessTelemetry =
+          (if (mappingCreateResult == MAPPING_CREATED) mapOf() else mapOf("mapping" to "initial-failure"))
```

Comment:
extra unnecessary `()` round the `if`?  Also why is the `also` block in the `apply` block, wouldn't it bettter be outside it?
Also might be cleaner to just define one variable for telemetry i.e.
```
  val mappingSuccessTelemetry = mapOf<String, String>(
      "dpsCaseNoteId" to caseNoteId.toString()
  ).apply {
    if (mappingCreateResult != MAPPING_CREATED) this  + ("mapping" to "initial-failure")
  }